### ISSUE TITLE
Make sure we reject everything in src/test/ui/parser/

### ISSUE THREAD

**Issue by Centril** [rust-lang-nursery/wg-grammar#11]
We'd like the grammar to reject everything inside https://github.com/rust-lang/rust/tree/master/src/test/ui/parser since those are parse-fail tests.

**Comment by eddyb** [comment]
We might want to look for hints like `// compile-flags: ` with `-Z parse-only` on the same line, accompanied by `ERROR` comments and/or a `stderr` file, instead of hardcoding a directory.

**Comment by Centril** [comment]
Yeah that might be more robust, but it also relies on `-Z parse-only` remaining the way in compile_test so you'll have to make sure that the rest of the compiler team don't sweep that out from under us without prior notice ;)

**Comment by petrochenkov** [comment]
All `-Z parse-only` are indeed in `ui/parser` currently, but `ui/parser` also contains `compile-pass` tests and some non-`-Z parse-only` tests doing parsing-during-expansion, so https://github.com/rust-lang-nursery/wg-grammar/issues/11#issuecomment-434895047 is certainly more robust.
